pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -302,6 +302,59 @@ type DashboardSpec struct {
 	// SSL determines whether SSL should be used
 	// +optional
 	SSL bool `json:"ssl,omitempty"`
+	// SSO determines whether to enable SSO for dashboard
+	// +optional
+	SSO SSOSpec `json:"sso,omitempty"`
+}
+
+type SSOSpec struct {
+	// Enabled determines whether to enable SSO
+	// +optional
+	Enabled bool `json:"enabled,omitempty"`
+	// If users that login through SSO should be auto created
+	// +optional
+	BaseURL string `json:"baseUrl,omitempty"`
+	// URL to IDP Metadata
+	// +optional
+	IDPMetadataURL string `json:"idpMetadataUrl,omitempty"`
+	// List of users to be created with a respective role from the admin side
+	// +optional
+	Users []UserRef `json:"users,omitempty"`
+	// IDP Attributes that should be used to get the username from the authentication response. Defaults to uid.
+	// +optional
+	IDPAttributes IDPAttributes `json:"idpAttributes,omitempty"`
+	// To be used when more than one entity id exists on the IdP metadata
+	// +optional
+	EntityID string `json:"entityID,omitempty"`
+	// Certificate that should be used by Ceph Dashboard for signing and encryption
+	// +optional
+	SPCert SecretKeyRef `json:"spCert,omitempty"`
+	// Private key that should be used by Ceph Dashboard for signing and encryption
+	// +optional
+	SPPrivateKey SecretKeyRef `json:"spPrivateKey,omitempty"`
```

Comment:
These two fields aren't implemented (yet). Let's remove them for now and add them in the future (make sure to re-generate the CRD definitions)